internal/resources/google/kms_crypto_key.go

```diff
@@ -0,0 +1,200 @@
+)
+
+type KMSCryptoKey struct {
+	Address                         string
+	Region                          string
+	VersionTemplate                 string
+	VersionTemplate0Algorithm       string
+	VersionTemplate0ProtectionLevel string
+	RotationPeriod                  string
+	KeyVersions                     *int64 `infracost_usage:"key_versions"`
+	MonthlyKeyOperations            *int64 `infracost_usage:"monthly_key_operations"`
+}
+
+var KMSCryptoKeyUsageSchema = []*schema.UsageItem{{Key: "key_versions", ValueType: schema.Int64, DefaultValue: 0}, {Key: "monthly_key_operations", ValueType: schema.Int64, DefaultValue: 0}}
+
+func (r *KMSCryptoKey) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+func (r *KMSCryptoKey) BuildResource() *schema.Resource {
+	region := r.Region
+
+	algorithm := "GOOGLE_SYMMETRIC_ENCRYPTION"
+	protectionLevel := "SOFTWARE"
+
+	if r.VersionTemplate != "" {
+		algorithm = r.VersionTemplate0Algorithm
+		protectionLevel = r.VersionTemplate0ProtectionLevel
+	}
+
+	var monthlyKeys *decimal.Decimal
+	if r.KeyVersions != nil {
+		monthlyKeys = decimalPtr(decimal.NewFromInt(*r.KeyVersions))
+	} else if r.RotationPeriod != "" {
+		rotationPeriod := (r.RotationPeriod)
+		rotation, err := strconv.ParseFloat(strings.Split(rotationPeriod, "s")[0], 64)
+
+		if err == nil {
+			monthlyKeys = decimalPtr(decimal.NewFromFloat(2592000.0 / rotation))
+		}
+	}
+
+	var monthlyKeyOperations *decimal.Decimal
+	if r.MonthlyKeyOperations != nil {
+		monthlyKeyOperations = decimalPtr(decimal.NewFromInt(*r.MonthlyKeyOperations))
+	}
+
+	var keyDescript = cryptoKeyDescription(algorithm, protectionLevel)
+	var operationDesctipt = keyOperationsDescription(algorithm, protectionLevel)
+
+	costComponents := []*schema.CostComponent{}
+
+	if strings.ToLower(keyDescript) == "hsm rsa 3072" || strings.ToLower(keyDescript) == "hsm rsa 4096" || strings.ToLower(keyDescript) == "hsm ecdsa p-256" || strings.ToLower(keyDescript) == "hsm ecdsa p-384" {
+		tierLimits := []int{2000}
+		var firstTierQty *decimal.Decimal
+		var tiers []decimal.Decimal
+		if monthlyKeys != nil {
+			tiers = usage.CalculateTierBuckets(*monthlyKeys, tierLimits)
+			firstTierQty = &tiers[0]
+		}
+
+		costComponents = append(costComponents, &schema.CostComponent{
+			Name:            "Key versions (first 2K)",
+			Unit:            "months",
+			UnitMultiplier:  decimal.NewFromInt(1),
+			MonthlyQuantity: firstTierQty,
+			ProductFilter: &schema.ProductFilter{
+				VendorName:    strPtr("gcp"),
+				Region:        strPtr(region),
+				Service:       strPtr("Cloud Key Management Service (KMS)"),
+				ProductFamily: strPtr("ApplicationServices"),
+				AttributeFilters: []*schema.AttributeFilter{
+					{Key: "description", ValueRegex: strPtr(fmt.Sprintf("/%s/i", keyDescript))},
+				},
+			},
+			PriceFilter: &schema.PriceFilter{
+				EndUsageAmount: strPtr("2000"),
+			},
+		})
+
+		if tiers != nil && tiers[1].GreaterThan(decimal.NewFromInt(0)) {
```

Comment:
```suggestion
		if len(tiers) > 1 {
```
Should we check that `tiers[1]` exists as above?

Reply:
Nice catch. Fixed